doc/source/whatsnew/v0.24.0.rst

```diff
@@ -994,6 +994,7 @@ update the ``ExtensionDtype._metadata`` tuple to match the signature of your
 - :meth:`Series.astype` and :meth:`DataFrame.astype` now dispatch to :meth:`ExtensionArray.astype` (:issue:`21185:`).
 - Slicing a single row of a ``DataFrame`` with multiple ExtensionArrays of the same type now preserves the dtype, rather than coercing to object (:issue:`22784`)
 - Added :meth:`pandas.api.types.register_extension_dtype` to register an extension type with pandas (:issue:`22664`)
+- Added :meth:`pandas.api.extensions.ExtensionArray.where` (:issue:`24077`)
```

Comment:
where-related changes are from #24114. e7538e67dfcd33fcce6d0f40fe4c6e54e5a63cad has the changes that'll stay here (moving PeriodArray.where up to DatetimeLikeArray.where).

There's still one where-related failure, but it's a different maybe bug(?) in internals. Still debugging.